HHH-20510 - Bring back hibernate-ucp module (#12587)
hibernate/hibernate-orm: documentation/src/main/asciidoc/userguide/chapters/jdbc/Database_Access.adoc

`hibernate.connection.isolation`:: Mapped to Agroal's `jdbcTransactionIsolation` setting. See <<ConnectionProvider support for transaction isolation setting>>.
`hibernate.connection.autocommit`:: Mapped to Agroal's `autoCommit` setting

[[database-connectionprovider-oracleucp]]
=== Using Oracle Universal Connection Pool

[IMPORTANT]
====
To use the Oracle Universal Connection Pool (aka UCP) integration, the application must include the `hibernate-ucp` module jar (as well as its dependencies) on the classpath.
====

> Hibernate also provides support for applications to use https://docs.oracle.com/en/database/oracle/oracle-database/26/jjucp/intro.html[Oracle Universal Connection Pool].

Reviewer: ```suggestion
Hibernate also provides support for applications using https://docs.oracle.com/en/database/oracle/oracle-database/26/jjucp/intro.html[Oracle Universal Connection Pool].
```

Set all of your UCP settings in Hibernate prefixed by `hibernate.ucp.` and this `ConnectionProvider` will pick them up, and pass them along to UCP.

> Additionally, this `ConnectionProvider` will pick up the following Hibernate-specific properties and map them to the corresponding UCP ones (any `hibernate.ucp.` prefixed ones have precedence):

Reviewer: ```suggestion
UCP settings may be supplied by prefixing the setting name with `hibernate.ucp.`.
Additionally, the UCP `ConnectionProvider` understands the following standard properties:
```

`hibernate.connection.url`:: Mapped to UCP `URL` setting
`hibernate.connection.username`:: Mapped to UCP `user` setting
`hibernate.connection.password`:: Mapped to UCP `password` setting
`hibernate.connection.isolation`:: Used to initialize the `Connection` retrieved from UCP. See <<ConnectionProvider support for transaction isolation setting>>.

> `hibernate.connection.autocommit`:: Used to initialize the `Connection` retrieved from UCP.

Reviewer: These are not the JPA standard property names. Might be worth also supporting `jakarta.persistence.jdbc.user` and friends.

Author: Very good remark, thanks!

> Any other settings prefixed with `hibernate.ucp.`:: Will have the `hibernate.ucp.` portion stripped and be passed to UCP (for example `hibernate.ucp.loginTimeout 5000`).

Reviewer: ```suggestion
Any other setting prefixed with `hibernate.ucp.`:: The `hibernate.ucp.` prefix is stripped and the remaining text is passed to UCP (for example, `hibernate.ucp.loginTimeout 5000`).
```

[IMPORTANT]
====
You can pass further properties to the `Connection` provided by UCP by using the `hibernate.ucp.connectionProperties` property in the following manner:

`hibernate.ucp.connectionProperties=oracle.jdbc.thinForceDNSLoadBalancing=true,oracle.jdbc.fanEnabled=true,oracle.jdbc.defaultConnectionValidation=SOCKET,oracle.jdbc.implicitStatementCacheSize=50,oracle.jdbc.loginTimeout=5000`

You can find the exhaustive list of the supported `Connection` properties https://docs.oracle.com/en/database/oracle/oracle-database/26/jajdb/oracle/jdbc/OracleConnection.html[here].

====

> The Hibernate property `hibernate.ucp.connectionFactoryClassName` can be used to choose between:

Reviewer: ```suggestion
The property `hibernate.ucp.connectionFactoryClassName` selects between:
```
